xray/segment.go

```diff
@@ -263,6 +270,12 @@ func NewSegmentFromHeader(ctx context.Context, name string, r *http.Request, h *
 // Close a segment.
 func (seg *Segment) Close(err error) {
 	seg.Lock()
+
+	// If segment is dummy we return
+	if seg.Dummy {
```

Comment:
Should this be placed before `seg.Lock()`? What happens if segment left locked because it's a dummy?

Reply:
Yes. This is a good point. We should unlock the segment before returning in the case where segment is dummy.